File: web/projects/portal/src/app/widget/asset-tree/load-asset-tree-nodes-event.ts
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { AssetEntry } from "../../../../../shared/data/asset-entry";

export class LoadAssetTreeNodesEvent {
   private targetEntry: AssetEntry;
   private expandedDescendants: LoadAssetTreeNodesEvent[] = [];
   private path: string[] = null;
   private index: number = 0;
   private scope: number = 0;
   private loadAll: boolean = false;

   getTargetEntry(): AssetEntry {
      return this.targetEntry;
   }

   setTargetEntry(value: AssetEntry) {
      this.targetEntry = value;
   }

   getExpandedDescendants(): LoadAssetTreeNodesEvent[] {
      return this.expandedDescendants;
   }

   setExpandedDescendants(value: LoadAssetTreeNodesEvent[]) {
      this.expandedDescendants = value;
   }

   getPath(): string[] {
      return this.path;
   }

   setPath(path: string[]) {
      this.path = path;
   }

   getScope(): number {
      return this.scope;
   }

   setScope(scope: number) {
      this.scope = scope;
   }

   getIndex(): number {
      return this.index;
   }

   setIndex(index: number) {
      this.index = index;
   }

   isLoadAll(): boolean {
      return this.loadAll;
   }

   setLoadAll(value: boolean) {
      this.loadAll = value;
   }
}

File: core/src/main/java/inetsoft/web/composer/model/LoadAssetTreeNodesEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import inetsoft.uql.asset.AssetEntry;
import org.immutables.value.Value;

import javax.annotation.Nullable;
import java.util.ArrayList;
import java.util.List;

@Value.Immutable
@JsonDeserialize(builder = LoadAssetTreeNodesEvent.Builder.class)
public abstract class LoadAssetTreeNodesEvent {
   @Nullable
   public abstract AssetEntry targetEntry();

   @Value.Default
   public List<LoadAssetTreeNodesEvent> expandedDescendants() {
      return new ArrayList<>();
   }

   @Nullable
   public abstract String[] path();

   @Value.Default
   public int index() { return 0; }

   @Value.Default
   public int scope() { return 0; }

   @Value.Default
   public boolean loadAll() { return false; }

   public static Builder builder() {
      return new Builder();
   }

   public static class Builder extends ImmutableLoadAssetTreeNodesEvent.Builder {
   }
}
